spacy/tests/doc/test_json_doc_conversion.py

```diff
@@ -70,3 +71,52 @@ def test_doc_to_json_span(doc):
     assert len(json_doc["spans"]) == 1
     assert len(json_doc["spans"]["test"]) == 2
     assert json_doc["spans"]["test"][0]["start"] == 0
+
+
+def test_json_to_doc(doc):
+    new_doc = Doc.from_json(doc.vocab, doc.to_json())
+    new_tokens = [token for token in new_doc]
+    assert new_doc.text == doc.text
+    assert len(new_tokens) == len([token for token in doc])
+    assert new_doc.text == doc.text == "c d e "
+    assert len(new_tokens) == 3
+    assert new_tokens[0].pos_ == "VERB"
+    assert new_tokens[0].tag_ == "VBP"
+    assert new_tokens[0].dep_ == "ROOT"
```

Comment:
Included in next commit. Using `new_tokens[0].head.idx == doc[0].head.idx` to compare the `head` info now, because `Token.__richcmp__()` also checks for equality of the heads' docs, which is not given in our case. Let me know if there's a better way to check for head equality :)